tests/app/legacy/dao/test_recipient_identifiers.py

```diff
@@ -0,0 +1,174 @@
+"""Tests for recipient identifiers DAO methods."""
+
+from typing import Any
+from unittest.mock import AsyncMock, patch
+from uuid import uuid4
+
+import pytest
+from sqlalchemy import Row, delete, select
+from sqlalchemy.exc import (
+    DataError,
+    IntegrityError,
+    InterfaceError,
+    OperationalError,
+    SQLAlchemyError,
+    TimeoutError,
+)
+
+from app.constants import IdentifierType
+from app.db.db_init import get_write_session_with_context, metadata_legacy
+from app.exceptions import NonRetryableError, RetryableError
+from app.legacy.dao.recipient_identifiers_dao import RecipientIdentifiersDao
+from app.legacy.v2.notifications.route_schema import RecipientIdentifierModel
+
+
+class TestLegacyRecipientIdentifiersDaoSet:
+    """Test class for LegacyRecipientIdentifiersDao set_recipient_identifiers."""
+
+    @pytest.mark.parametrize(
+        ('id_type', 'id_value'),
+        [
+            (IdentifierType.BIRLSID, '12345'),
+            (IdentifierType.EDIPI, '12345'),
+            (IdentifierType.ICN, '1234567890V123456'),
+            (IdentifierType.PID, '12345'),
+        ],
```

Comment:
Did we want this to be a complete list? There's no VAPROFILEID. It should all still work the same, just pointing out that we are missing one.